### PR TITLE
WIP: generalize stimulus to allow selection of state variables

### PR DESCRIPTION
**Pull request by i-Zaak**
Most conservative generalization is to allow specification of variables, upon which the stimulus is applied (currently these are the coupling variables). Next we might also allow e.g. different scaling of the stimulus or completely different stimulus for different state variables. 

Related to https://req.thevirtualbrain.org/browse/TVB-1815

### CODE REVIEW

**Comment by maedoc** [comment]
👍 this is good

> different scaling of the stimulus or completely different stimulus for different state variables.

this would be good and rather trivial if not for the well-intentioned `.reshape((1, -1, 1))`.  I see this largely as an API design failure, and we can do better in near future.

File: scientific_library/tvb/simulator/models/base.py [hunk 1]
```diff
@@ -91,6 +92,8 @@ def configure(self):
         "Configure base model."
         for req_attr in 'nvar number_of_modes cvar'.split():
             assert hasattr(self, req_attr)
+        if self.stvar is None:
+            self.stvar = self.cvar
```
Comment (maedoc) [inline scientific_library/tvb/simulator/models/base.py]: let's `self.cvar.copy()` just in case someone decides to modify `cvar` in place

**Comment by maedoc** [comment]
do you have any tests for this?

**Comment by i-Zaak** [comment]
> do you have any tests for this?

Done.

**Comment by maedoc** [comment]
Sorry for the delay, I'm trying to setup a CI thing and use this PR as the guinea pig.  It all looks good otherwise.

**Comment by i-Zaak** [comment]
:+1: :guinea: :pig2:

**Comment by maedoc** [comment]
```bash
git remote add upstream https://github.com/the-virtual-brain/tvb-root
git fetch -a
git checkout stimulus-generalization
git rebase remotes/upstream/master
git push -f origin stimulus-generalization
```
@i-Zaak plz?

**Comment by i-Zaak** [comment]
@maedoc done.

**Comment by maedoc** [comment]
hm, CI didn't seem to run here? maybe the rebase didn't go as planned.  It any cases, tests merge, let's merge

**Pull request merged**